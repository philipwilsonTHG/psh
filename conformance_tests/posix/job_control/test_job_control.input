# Test job control commands (jobs, fg, bg)
# PSH supports job control with jobs, fg, bg commands (v0.9.0 feature)

# Test jobs command with no jobs
echo "Testing jobs with no active jobs:"
jobs 2>/dev/null || echo "Jobs command completed (no jobs)"

# Test background job and jobs command
echo "Testing jobs command with background job:"
true &
job_pid=$!
echo "Background job started"
jobs 2>/dev/null || echo "Jobs command executed"
wait $job_pid
echo "Background job completed"

# Test jobs command after completion
echo "Testing jobs after job completion:"
jobs 2>/dev/null || echo "Jobs command executed (no active jobs)"

# Test multiple background jobs
echo "Testing multiple background jobs:"
true &
pid1=$!
true &
pid2=$!
true &
pid3=$!
echo "Started 3 background jobs"
jobs 2>/dev/null || echo "Jobs command with multiple jobs executed"
wait $pid1
wait $pid2
wait $pid3
echo "All background jobs completed"

# Test job control in pipeline context
echo "Testing job control in pipeline:"
(echo "test" | cat) &
pipe_pid=$!
echo "Pipeline job started"
wait $pipe_pid
echo "Pipeline job completed"

# Test job specification patterns
echo "Testing job specifications:"
true &
bg_pid=$!
jobs 2>/dev/null || echo "Job specification test executed"
wait $bg_pid
echo "Job specification test completed"