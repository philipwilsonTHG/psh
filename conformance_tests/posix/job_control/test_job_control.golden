Testing jobs with no active jobs:
Testing jobs command with background job:
Background job started
Background job completed
Testing jobs after job completion:
Testing multiple background jobs:
Started 3 background jobs
All background jobs completed
Testing job control in pipeline:
Pipeline job started
Pipeline job completed
Testing job specifications:
Job specification test completed
